pkg/thread/thread.go

```diff
@@ -18,11 +18,12 @@ import (
 //    ....
 //
 type Thread struct {
-	log      logrus.FieldLogger
-	exec     func()
-	done     chan struct{}
-	name     string
-	interval time.Duration
+	log              logrus.FieldLogger
+	exec             func()
+	done             chan struct{}
+	name             string
+	interval         time.Duration
+	lastRunStartedAt time.Time
```

Comment:
lets init the time in the New Call, we don't know how long the first call will run so we can get into a state that we started the thread the first exec takes 1 minute but the health check will already run and see that the time is zero and reboot the service.

Reply:
We are now setting the `lastRunStartedAt` right before the `exec` so, I think it's ok to skip with the `IsZero` check. Both work, as in we could init it and remove the check but as it is it should have the same effect. Thoughts?

Reply:
Actually it should be set in the Start function.

Reply:
I changed the var to lastRunStartedAt due to that we will set it before the operation starts.

Reply:
It can create a different bug that if we never start the monitor health will report that everything is ok. 
it's happening right now as well but at least if we already adding something that validate it then we should cover this case as well. 
So even if you set the time at New we will know that something is wrong.

Reply:
I think that we should set it in the start just before the `go` call

Reply:
Agree, fixed